SPECS/curl/curl.spec

```diff
@@ -98,6 +87,9 @@ rm -rf %{buildroot}/*
 %{_libdir}/libcurl.so.*
 
 %changelog
+* Fri May 07 2021 Daniel Burgener <daburgen@microsoft.com> - 7.76.0-2
```

Comment:
Can you give more details on the circular dependency? Asking because recently, in the dev branch, we fixed a problem where our package graph was not taking the `RUN_CHECK` Make variable into account when parsing requirements from spec files.

Link: https://github.com/microsoft/CBL-Mariner/pull/945

Would love to get more details on your problem and see if that fix would work here, since this is a fairly core package to us and it'd be preferable to not remove the check section. It'd also help me prioritize porting this change back to 1.0-dev.

Reply:
I've been trying to reproduce this all day with little success.  I suppose it's possible that something has changed upstream already which addresses it.  I'm going to try a full end to end pipeline build with this change removed and see what happens there.

The note I have from my git commit is:

The curl check code causes a circular dependency with shadow-utils (and
    sudo) once the new libsemanage shadow-utils dependency is added (which is
    required for SELinux support).  Comment the check out to see if that
    fixes the issues before further investigation into correct fix.

Unfortunately, that doesn't seem to provide the level of detail that I need to give a perfect answer to your question.  We'll see if an overnight pipeline run gives me a repro and I'll let you know tomorrow.

Reply:
Okay, I was able to get a repro:

ERRO[0035] Unfixable circular dependency of length 142193:	{audit-libs-3.0-6.cm1-RUN<Meta>} --> {audit-libs-3.0-6.cm1-BUILD<Build>} --> {openldap-2.4.57-2.cm1-RUN<Meta>} --> {openldap-2.4.57-2.cm1-BUILD<Build>} --> {cyrus-sasl-2.1.27-4.cm1-RUN<Meta>} --> {systemd-239-37.cm1-RUN<Meta>} --> {systemd-239-37.cm1-BUILD<Build>} --> {cryptsetup-devel-2.3.3-2.cm1-RUN<Meta>} --> {cryptsetup-2.3.3-2.cm1-RUN<Meta>} --> {cryptsetup-libs-2.3.3-2.cm1-RUN<Meta>} --> {cryptsetup-libs-2.3.3-2.cm1-BUILD<Build>} --> {device-mapper-devel-2.03.05-5.cm1-RUN<Meta>} --> {device-mapper-2.03.05-5.cm1-RUN<Meta>} --> {device-mapper-2.03.05-5.cm1-BUILD<Build>} --> {systemd-bootstrap-devel-239-34.cm1-RUN<Meta>} --> {systemd-bootstrap-239-34.cm1-RUN<Meta>} --> {systemd-bootstrap-239-34.cm1-BUILD<Build>} --> {meson-0.56.0-1.cm1-RUN<Meta>} --> {meson-0.56.0-1.cm1-BUILD<Build>} --> {ninja-build-1.8.2-3.cm1-RUN<Meta>} --> {ninja-build-1.8.2-3.cm1-BUILD<Build>} --> {gtest-devel-1.8.1-5.cm1-RUN<Meta>} --> {gtest-devel-1.8.1-5.cm1-BUILD<Build>} --> {cmake-3.17.3-2.cm1-RUN<Meta>} --> {cmake-3.17.3-2.cm1-BUILD<Build>} --> {curl-devel-7.76.0-2.cm1-RUN<Meta>} --> {curl-7.76.0-2.cm1-RUN<Meta>} --> {curl-7.76.0-2.cm1-BUILD<Build>} --> {sudo-1.9.5p2-2.cm1-RUN<Meta>} --> {shadow-utils-4.6-11.cm1-RUN<Meta>} --> {shadow-utils-4.6-11.cm1-BUILD<Build>} --> {libsemanage-devel-2.9-4.cm1-RUN<Meta>} --> {libsemanage(x86-64)-2.9-4.cm1-RUN<Meta>} --> {audit-libs-3.0-6.cm1-RUN<Meta>} 
ERRO[0035] Error found while resolving package dependency cycles: cycle contains build dependencies, unresolvabl

That's just one example.  The full pipeline log is here: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=93957&view=logs&j=75cba72d-09cb-503a-3927-4ab61ac4a179&t=e3d06be3-74ca-556e-743b-b543e06aae03

The core issue as I see it is that libsemanage has a lot of dependencies, and this commit introduces a shadow-utils dependency on libsemanage, which is necessary to set user login contexts with SELinux.  Curl is pulled in by cmake, which is in turn a dependency for a lot of packages.  I'm not sure what the cause of the cmake dependency on curl is and whether there is some way to break the cycle there.  Comparison with Fedora and Red Hat shows that they have no dependency on shadow-utils or sudo in the curl spec file.

If your change would fix this problem retaining the check code, that would be great.